### PR TITLE
Add -b/--build-vendor flag to add-on build scripts, disable auto-build.

### PR DESCRIPTION
**Pull request by Osmose**
I got tired of waiting for NPM to build when 99% of the time I don't need it to auto-build.

I, of course, am the one who made the scripts auto-build in the first place.

### CODE REVIEW

**Comment by Osmose** [comment]
TC failure seems to be a Rust-related issue @mythmon and I talked about earlier. I'll rebase once a fix lands.

**Comment by Osmose** [comment]
Rebased, build is fixed.

File: docs/dev/recipe-client-addon/workflow.rst [hunk 1]
```diff
@@ -71,6 +71,9 @@ files:
    cd recipe-client-addon
    ./bin/make-xpi.sh # Creates /recipe-client-addon/shield-recipe-client.xpi
 
+Pass the ``-b``/``--build-vendor`` flag to auto-build vendor dependencies before
+creating the XPI.
```
Comment (mythmon) [inline docs/dev/recipe-client-addon/workflow.rst]: I think that the default value here should be to build the vendor library, or to autodetect if it hasn't ever been built. I'd be ok if we didn't rebuild by default. This is mostly for ease of  use for new or infrequent contributors.

**Comment by Osmose** [comment]
Updated; the scripts now auto-build `vendor` if it doesn't yet exist.

**Pull request merged**